memcache/memcache.go

```diff
@@ -89,7 +89,7 @@ func legalKey(key string) bool {
 		return false
 	}
 	for i := 0; i < len(key); i++ {
-		if key[i] <= ' ' || key[i] > 0x7e {
+		if key[i] <= ' ' || key[i] == 0x7f {
```

Comment:
(and unless I'm missing something, this entire malformed key checking only extends to the ASCII space, i.e. it shouldn't matter for say UTF-8)